### ISSUE TITLE
Facebook URL 

### ISSUE THREAD

**Issue by RingoTheDog** [rg3/youtube-dl#5120]
URL: https://www.facebook.com/michelle.cote.90/posts/10153188171507704

C:\>youtube-dl -v https://www.facebook.com/michelle.cote.90/posts/10153188171507704
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', 'https://www.facebook.com/michelle.cote.90/pos
ts/10153188171507704']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2015.03.03.1
[debug] Python version 2.7.8 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-40824-g31dfe20
[debug] Proxy map: {}
[generic] 10153188171507704: Requesting header
WARNING: Falling back on generic information extractor.
[generic] 10153188171507704: Downloading webpage
[generic] 10153188171507704: Extracting information
ERROR: Unsupported URL: https://www.facebook.com/michelle.cote.90/posts/10153188
171507704
Traceback (most recent call last):
  File "youtube_dl\extractor\generic.pyo", line 775, in _real_extract
  File "youtube_dl\utils.pyo", line 1506, in parse_xml
  File "xml\etree\ElementTree.pyo", line 1300, in XML
  File "xml\etree\ElementTree.pyo", line 1642, in feed
  File "xml\etree\ElementTree.pyo", line 1506, in _raiseerror
ParseError: not well-formed (invalid token): line 31, column 3156
Traceback (most recent call last):
  File "youtube_dl\YoutubeDL.pyo", line 648, in extract_info
  File "youtube_dl\extractor\common.pyo", line 275, in extract
  File "youtube_dl\extractor\generic.pyo", line 1226, in _real_extract
UnsupportedError: Unsupported URL: https://www.facebook.com/michelle.cote.90/pos
ts/10153188171507704

(Similar result with HTTP:// as opposed to HTTPS://)

Thanks 
Ringo

**Comment by naglis** [comment]
FWIW, if you click on/copy the link of the video or it's title on the post, you can get the video's URL which works fine with youtube-dl, in this case: 
```
https://www.facebook.com/video.php?v=10153188168547704&fref=nf
```

**Comment by RingoTheDog** [comment]
Similar example:

These URLs do not work (falls back to generic extractor)
https://www.facebook.com/ChristyClarkForBC/videos/10153870694020942/
https://www.facebook.com/ChristyClarkForBC/videos/vb.22819070941/10153870694020942/?type=2&theater

This URL does work (same video)
https://www.facebook.com/video.php?v=10153870694020942&fref=nf

**Comment by yan12125** [comment]
Thanks @RingoTheDog. Such URLs are directly supported by youtube-dl in the next version. Please open a new issue the next time. They are different issues.

**Comment by dstftw** [comment]
This has been fixed in youtube-dl 2015.04.26. See our [FAQ](https://yt-dl.org/update) if you have problems updating.
